core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,729 @@ The document defines storage capability requirements in the following areas:
 
+[[R282]](#r282) Testability: 
+
+Preconditions: 
+
+* A BPI Storage e.g. the open-source [PostgreSQL](https://www.postgresql.org/) data base deployed in a BPI test environment with a BPI client and a BPI server
+* An authentication service to be used with the BPI Storage such as [[OAuth-2.0](#oauth-20)] between client and server.
+
+Test steps:
+
+1. Create a BPI Subject within the BPI with a public-private cryptographic key pair based on one of the cryptographic algorithms supported by the authentication service chosen for the BPI Subject
+2. Create an authentication policy for the BPI Storage supported by each of the authemtication services chosen for the test  
+2. Attempt to authenticate the BPI Subject with the BPI Storage using each of the identified external authentication services for the test.
+3. Verify that the authentication of the BPI Subject is successful and the BPI Subject is granted the appropriate access to the BPI Storage as per the test authentication policy.
+4. Repeat the above steps for each supported external authentication service.
+
+Test pass criteria:
+
+The test will pass if one or more of the external authentication services successfully authenticate a BPI Subject to the BPI Storage and the BPI Subject is granted the type of access to the BPI Storage as defined in the test policy. 
+
+Note that the test policy could be any combination of read and write access to the BPI Storage.
+
 #### **[R283]**	
 BPI Storage MUST support roles & access management.
 
+[[R283]](#r283) Testability: 
+
+Preconditions: 
+
+* Identify roles and levels of access required that need to be supported by the BPI Storage, based on the business requirements of the test use case.
+* A BPI test system is up and running.
+* Two or more BPI Subjects and BPI Subject Accounts created in the BPI
+
+Test steps:
+
+1. Create two or more test BPU Subjects and assign them to the identified roles and the level of access specified for each role.
+2. Attempt to perform various operations on the BPI Storage using each test BPI Subject.
+3. Verify that the operations are allowed or denied based on the access permissions associated with each of the BPI Subjects' role.
+4. Attempt to perform operations that require different access permissions than the test BPI Subjects' role.
+5. Verify that such operations are denied and an appropriate error message is displayed.
+6. Repeat the above steps for each test role.
+
+Test pass criteria:
+
+* The test will pass if all test roles are successfully tested, and the test BPI Subjects are able to perform operations only as allowed by their role's permissions. 
+* Additionally, attempting operations that require different permissions than the test BPI Subjects' roles must be denied, and appropriate error messages must be displayed.
+
 #### **[R284]**	
 BPI Storage MUST support policy management.
 
+*Examples of such policies are rules governing creation, modification, deletion, and retention of data.* 
```

Comment:
```suggestion
BPI Storage MUST support policy management, at least enabling the creation, modification and removal of rules governing creation, modification, deletion, and retention of data.
```
This change provides an example minimal functional requirement. It can be tested, and it is a useful set of policies that can be managed. There is no constraint on what else might be implemented, and multiple specific implementation strategies can work (e.g. build a UI directly in the tool, import configuration files defined and created somewhere else, enable ABAC or RBAC control over who can perform these tasks, ...)

Reply:
@chaals it was intentionally left as an example because an append-only BPI storage, such as IPFS, a perfectly valid choice, is append-only by design, without the need for extra policies as would be required for e.g. PostGres. Therefore, embedding specific data policies as requirements constrains the solution space too much.  I would very much prefer to leave the examples as is and allow more freedom as to the type of policies e.g. each a write creating a zkp of correct write execution.

Reply:
The problem with that is as a conscientious implementor, you are making it hard for me to know when I am done. And as someone who just wants to tick the box, I'll implement the ability to turn off some flag, call it a policy, and claim I am done.

No interoperability in practice, and a requirement that frustrates people trying to do the right thing.

Setting a minimum bar is helpful. I agree with you 100% that constraining people to some minimal level is just a bad idea.

Reply:
@chaals policy management needs to be defined -- just CRUD for policies is what this means. Lemme add that. That should clear it up. Supporting role and access policies is an earlier requirement.